controllers/cloudinit/cloudinit_test.go

```diff
@@ -20,250 +20,77 @@ import (
-			WriteFiles:          nil,
-			Users:               nil,
-			NTP:                 nil,
-			DiskSetup: &bootstrapv1.DiskSetup{
-				Partitions: []bootstrapv1.Partition{
-					{
-						Device:    "test-device",
-						Layout:    true,
-						Overwrite: pointer.BoolPtr(false),
-						TableType: pointer.StringPtr("gpt"),
-					},
-				},
-				Filesystems: []bootstrapv1.Filesystem{
-					{
-						Device:     "test-device",
-						Filesystem: "ext4",
-						Label:      "test_disk",
-						ExtraOpts:  []string{"-F", "-E", "lazy_itable_init=1,lazy_journal_init=1"},
-					},
-				},
-			},
-			Mounts: []bootstrapv1.MountPoints{
-				{"test_disk", "/var/lib/testdir"},
-			},
-		},
-		Certificates:         secret.Certificates{},
-		ClusterConfiguration: "my-cluster-config",
-		InitConfiguration:    "my-init-config",
+	for _, f := range expectedCommands {
+		g.Expect(string(out)).To(ContainSubstring(f))
 	}
-
-	out, err := NewInitControlPlane(cpinput)
-	g.Expect(err).NotTo(HaveOccurred())
-
-	expectedDiskSetup := `disk_setup:
-  test-device:
-    table_type: gpt
-    layout: true
-    overwrite: false`
-	expectedFSSetup := `fs_setup:
-  - label: test_disk
-    filesystem: ext4
-    device: test-device
-    extra_opts:
-      - -F
-      - -E
-      - lazy_itable_init=1,lazy_journal_init=1`
-	expectedMounts := `mounts:
-  - - test_disk
-    - /var/lib/testdir`
-
-	g.Expect(string(out)).To(ContainSubstring(expectedDiskSetup))
-	g.Expect(string(out)).To(ContainSubstring(expectedFSSetup))
-	g.Expect(string(out)).To(ContainSubstring(expectedMounts))
 }
 
-func TestNewJoinControlPlaneAdditionalFileEncodings(t *testing.T) {
+func TestNewJoinControlPlaneCommands(t *testing.T) {
 	g := NewWithT(t)
 
 	cpinput := &ControlPlaneJoinInput{
-		BaseUserData: BaseUserData{
-			Header:              "test",
-			PreKubeadmCommands:  nil,
-			PostKubeadmCommands: nil,
-			AdditionalFiles: []bootstrapv1.File{
-				{
-					Path:     "/tmp/my-path",
-					Encoding: bootstrapv1.Base64,
-					Content:  "aGk=",
-				},
-				{
-					Path:    "/tmp/my-other-path",
-					Content: "hi",
-				},
-			},
-			WriteFiles: nil,
-			Users:      nil,
-			NTP:        nil,
-		},
-		Certificates: secret.Certificates{},
-		JoinToken:    "my-bootstrap-token",
-	}
-
-	for _, certificate := range cpinput.Certificates {
-		certificate.KeyPair = &certs.KeyPair{
-			Cert: []byte("some certificate"),
-			Key:  []byte("some key"),
-		}
+		ControlPlaneEndpoint: "lb-ep",
+		JoinToken:            "my_join_token",
+		Version:              "v1.24.3",
+		IPOfNodeToJoin:       "1.2.3.4",
+		PortOfNodeToJoin:     "25000",
 	}
 
 	out, err := NewJoinControlPlane(cpinput)
 	g.Expect(err).NotTo(HaveOccurred())
 
-	expectedFiles := []string{
-		`-   path: /tmp/my-path
-    encoding: "base64"
-    content: |
-      aGk=`,
-		`-   path: /tmp/my-other-path
-    content: |
-      hi`,
+	expectedCommands := []string{
+		`snap install microk8s --classic --channel=1.24`,
+		`--token my_join_token`,
```

Comment:
if it matters at all: should the token be specified in the `join` command instead? Just like below